File: src/main/java/com/aca/fitnessapp/controller/StackController.java
package com.aca.fitnessapp.controller;

import java.util.List;

import com.aca.fitnessapp.model.ClientStack;
import com.aca.fitnessapp.service.StackService;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/stacks")
public class StackController {

	private StackService service = new StackService();
	
	@GET
	@Path("/client")
	@Produces(MediaType.APPLICATION_JSON)
	public List<ClientStack> getExercises() {
		return service.getClientStacks();
	}
}

File: src/main/webapp/js/stackController.js



(function() {
	let fitnessApp = angular.module('fitnessApp');
	
	fitnessApp.controller('stackController', function($scope, $http) {
		
		
	$scope.getClientStacks = function() {
		$http.get("/fitnessapp/webapi/stacks/client")
		.then(function(response) {
			$scope.client_stacks = response.data;
		}, function(response) {
			console.log('error http GET stacks/client: ' + response.status);
		});
	}
	$scope.getClientStacks();
	
	});
})();